homeassistant/scripts/check_config.py

```diff
@@ -252,7 +252,7 @@ def dump_dict(layer, indent_count=3, listi=False, **kwargs):
     """
     def sort_dict_key(val):
         """Return the dict key for sorting."""
-        key = str.lower(val[0])
+        key = str.lower(val[0]) if isinstance(val[0], str) else str(val[0])
```

Comment:
Yeah! That would be a nicer solution, but then it will always convert even if it is a string (witch it is in 99% of the cases). This is though just a function used with testing. So depends if nice code or speed is needed? ;)